applications/tari_mining_node/src/difficulty.rs

```diff
@@ -76,7 +76,11 @@ impl BlockHeaderSha3 {
     }
 
     pub fn set_timestamp(&mut self, timestamp: u64) {
-        self.timestamp = timestamp;
+        // if the timestamp has been advanced by the base_node due to the median time we should not reverse it but we
+        // should only change the timestamp if we move it forward.
+        if timestamp > self.timestamp {
```

Comment:
nit: I'd rather go for putting the check outside the function call (it's only used in one place anyhow). It isn't expected for `set_timestamp` not to set the timestamp. Alternatively, name the function `set_forward_timestamp` (or something) and explain in a docstring that is only sets the timestamp forward from what it is. This is a nit so feel free to ignore.

Reply:
I agree good idea